sklearn/utils/extmath.py

```diff
@@ -361,7 +361,10 @@ def randomized_svd(M, n_components, n_oversamples=10, n_iter=2,
     U = np.dot(Q, Uhat)
 
     if flip_sign:
-        U, V = svd_flip(U, V)
+        if not transpose:
+            U, V = svd_flip(U, V)
+        else:
+            U, V = svd_flip(U, V, u_based_decision=False)
```

Comment:
Maybe add a comment after the else saying in essence that we need u_based_decision=False so that the flipping is actually based on u after transpose.